### DISCUSSION TITLE
Advice on editing the hover height?

### DISCUSSION THREAD

**Discussion by Kiesmith** [bitcraze/discussions#530] [Q&A]
I looked into App_push_demo: [https://github.com/bitcraze/crazyflie-firmware/blob/master/examples/demos/app_push_demo/src/push.c](url)

I recoded my own version a little bit and removed the statements so I do not need to wave my hand over for it to fly etc. it now automatically raises as soon as programmed which is what I want, but I am trying to change the height at that it can reach and I am rather unsure on the design of this code. I tried changing the height as I noticed **position.z** must determine the hover height, but it does not really change at all. I changed **height** too which is a variable to position.z  also. it seems to only hover at a height of 7 - 8cm. I tried changing random variables to see if I could make it go higher but it doesn't seem to.

Is there anything I need to consider in when wanting the drone to hover higher, like 1 meter or 2? Are there easier examples to work with for this matter? I would really like to know so I can stretch my drone's capabilities.

If there are more things to consider than raising the height, what would they be and why? 

Thankyou very much, 
Kieran.

**Comment by ataffanel** [comment]
You can set the height of the push demo by modifying the `height_sp` variable: https://github.com/bitcraze/crazyflie-firmware/blob/9d4323f7f912618ddecea6097d7aee6b1c95995c/examples/demos/app_push_demo/src/push.c#L81, it is currently set to a height of 20cm.

The flow deck uses the down facing ranging sensor for height sensing, so the main thing to look at is that if it senses anything else than the floor, the Crazyflie will jump up to stay at the set height above the detected object. This is not a big problem at the original 20cm setpoint but if you fly at 1m while interacting with the drone it can make it crash if it detects you or other objects like when pushed above a table.

Make sure you have a flow-deck V2: the height sensor should have a 'gold' color. The flow sensor V1 was limited to about 1M range, the V2 will have no problem sensing up to the ceiling.
